api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -15,31 +15,62 @@
  */
 package io.cloudevents;
 
+import com.fasterxml.jackson.databind.JsonNode;
+
 import java.util.Map;
 import java.util.Optional;
 
 /**
  * An abstract event envelope
- * @param <A> The attributes type
- * @param <T> The 'data' type
  * @author fabiojose
+ * @author slinkydeveloper
  */
-public interface CloudEvent<A extends Attributes, T> {
+public interface CloudEvent {
 
     /**
      * The event context attributes
      */
-    A getAttributes();
+    Attributes getAttributes();
```

Comment:
FYI https://github.com/cloudevents/sdk-java/issues/118

Reply:
cc @n3wscott

Reply:
I think the solutions here are two:

* CloudEvent is implemented once (`impl.CloudEventImp`) and different attributes versions are added through composition of objects.
* CloudEvent is implemented multiple times, one for each event version.

The reason why i went with the first alternative is to simplify the implementation of json marshal/unmarshal, let's review this comment later when this pr is ready.